### PR TITLE
Uploading dev files for git script

### PR DESCRIPTION
**Pull request by andrewtpoe**
I had to delete the other repo with the git script. Things went out of sync and then went haywire. Here is the new script with install instructions and functions in the README. When installed correctly you should just type < tiy new_repo > in the terminal.

@rpearce - I know you had a clean up function that you asked for in the program, but I don't know exactly what you wanted in it.

@mcmillanator - feel free to fork and add any features you would like

@seanmpage - I have no idea what happened with your connections and the whole forking deal we did. My stuff went way out of wack and the only thing I knew to do was delete and start over.

Everybody else - Please add issues for any additional functions you might like to see. I'd love to build this thing up and make it super easy to use.

### CODE REVIEW

**Comment by rpearce** [comment]
putting :eyes: on this now

File: tiyhw.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  # Check for homebrew
+  if test ! $(which brew); then
+    echo "Installing homebrew..."
+    ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+  fi
+
+  # Check for hub
+  if test ! $(which hub); then
+    echo "Installing hub..."
+    brew istall hub
```
Comment (rpearce) [inline tiyhw.sh]: `install`, not `istall`

File: tiyhw.sh [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  # Intro and explain the script.
+  echo "This function creates a new repository on your computer and on GitHub."
+  echo ""
+
+  # Gather the name for the folder and the repo
+  read -p "Enter a new directory name: " folder
+  echo ""
+
+  # See if the user would like to exit this function.
+  if [ "$folder" != "end" ]; then
```
Comment (rpearce) [inline tiyhw.sh]: does this allow for someone to exit the prompt and whatnot? If so, `exit` might be more appropriate

File: tiyhw.sh [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  read -p "Enter a new directory name: " folder
+  echo ""
+
+  # See if the user would like to exit this function.
+  if [ "$folder" != "end" ]; then 
+    # Add in error checking to determine if this folder already exists. 
+    # Proceed if no folder, display error and exit if folder does exist.
+    if [[ ! -e $folder ]]; then
+      # Create the folder and make it the focus.
+      mkdir -p $folder
```
Comment (rpearce) [inline tiyhw.sh]: `-p` is only used when the folder may/may not exist. If that will never be the case here, the `-p` doesn't belong

File: tiyhw.sh [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    echo ""
+  fi
+  unset folder
+}
+
+
+# This method sets the branch variable to the name of the users currently 
+# active git branch.
+function determine_branch {
+  if branch=$(git symbolic-ref --short -q HEAD)
```
Comment (rpearce) [inline tiyhw.sh]: this is pretty good. going to steal it

File: tiyhw.sh [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  new_repo
+
+elif [ "$1" = "is_working" ]; then
+  echo "The tiy script is working correctly. Congratulations!"
+
+elif [ "$1" = "first_run" ]; then
+  :
+
+else
+  echo "tiy does not have a function by that name."
```
Comment (rpearce) [inline tiyhw.sh]: wouldn't these `tiy`s be `tiyhw`s?

Comment (andrewtpoe) [inline tiyhw.sh]: The correct terminology here depends on the alias that the user puts in their Bash Profile. I'm actually using easyg as mine.

Comment (rpearce) [inline tiyhw.sh]: true that

On Tue, Jun 30, 2015 at 4:04 PM, Andrew T. Poe notifications@github.com
wrote:

> In tiyhw.sh
> https://github.com/andrewtpoe/git_automator/pull/1#discussion_r33617097:
> 
> > +#
> > +###############################################################################
> > +
> > +system_check
> > +
> > +if [ "$1" = "new_repo" ]; then
> > -  new_repo
> >   +
> >   +elif [ "$1" = "is_working" ]; then
> > -  echo "The tiy script is working correctly. Congratulations!"
> >   +
> >   +elif [ "$1" = "first_run" ]; then
> > -  :
> >   +
> >   +else
> > -  echo "tiy does not have a function by that name."
> 
> The correct terminology here depends on the alias that the user puts in
> their Bash Profile. I'm actually using easyg as mine.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/andrewtpoe/git_automator/pull/1/files#r33617097.

## 

Robert Pearce
http://www.robertwpearce.com
https://www.linkedin.com/in/robertwpearce
m-US: 1.843.568.9080
m-UK: +44 7477 424374

**Comment by andrewtpoe** [comment]
@rpearce: 

I figured out the curl commands, although the urls will need to be updated when this is merged into the master branch. 

Using the curl command in the README runs the install script, which in turn does the following: 
- Downloads and installs the easy_git script in the user's bin folder 
- Calls the first_run function, which should download and install any missing dependencies the user may have. 
- Adds the alias into the user's bash_profile and reloads the bash_profile, thus making the functions immediately accessible. 

Functions are now called like this: 

``` bash
easyg <Function_name>
```

Available functions can be found in the README.

Would appreciate a review.

**Comment by rpearce** [comment]
There should be some way to see what commands are available, but that is for another PR.

**Comment by rpearce** [comment]
:+1:

**Comment by rpearce** [comment]
`-bash: alias: /Users/rpearce/bin/easy_git.sh: not found`

**Comment by rpearce** [comment]
is that a lingering thing?

**Comment by rpearce** [comment]
I believe so. I just removed it an no probs

**Comment by rpearce** [comment]
:+1:

**Comment by andrewtpoe** [comment]
I suspect that alias was there from when you installed the script using the earlier version.

I will merge this into master.

**Pull request merged**